### PR TITLE
Add http-hello-world example for C

### PR DESCRIPTION
**Pull request by cdmurph32**
## Feature or Problem
Wasmcloud should have a basic C example.

## Related Issues
N/A

## Release Information
next

## Consumer Impact
It's just a new example.

### Unit Test(s)
None

### Acceptance or Integration
None

### Manual Verification
Build with `make` and run with `wash app deploy wadm.yaml`

### CODE REVIEW

File: examples/c/components/http-hello-world/wasmcloud.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name = "http-hello-world"
+version = "0.1.0"
+language = "c"
+type = "component"
+
+[component]
+wit_world = "hello"
```
Comment (brooksmtownsend) [inline examples/c/components/http-hello-world/wasmcloud.toml:7]:
```suggestion
[component]
build_command = "make http_hello_world.wasm"
build_artifact = "http_hello_world.wasm"
wit_world = "hello"
```

This small change should let folks run `wash build` to build and sign it!

File: examples/c/components/http-hello-world/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- `wasmtime` >=19.0.0 (if running with wasmtime)
+- [wasi-sdk](https://github.com/WebAssembly/wasi-sdk)
+- [wit-bindgen](https://github.com/bytecodealliance/wit-bindgen)
+
+## Building
+
+The wash command does not yet support C for building. The variable `WASI_SDK_DIR` defaults to `/opt/wasi-sdk`.
+```bash
+make
+```
```
Comment (brooksmtownsend) [inline examples/c/components/http-hello-world/README.md:18]: When making the change to run `wash build`, just note the WASI SDK as a dependency

File: examples/c/components/http-hello-world/Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Build zlib, then pdfio, and finally the pdf_text wasm.
```
Comment (brooksmtownsend) [inline examples/c/components/http-hello-world/Makefile:1]: nit: looks like this comment can be removed / updated

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Hey @cdmurph32 thank you so much for the patience here, I just have one request to make this work with `wash build` and then it'll be good to go! Thank you for this example. Let me know if my suggestion doesn't work

File: examples/c/components/http-hello-world/wasmcloud.toml (hunk 1 above)
Comment (cdmurph32) [reply examples/c/components/http-hello-world/wasmcloud.toml:7]: Awesome!

**Review by brooksmtownsend — APPROVED** [review]
Looks great @cdmurph32 !

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**